### DISCUSSION TITLE
Disabled Form field will not be saved

### DISCUSSION THREAD

**Discussion by mrahmadt** [filamentphp/filament#8419] [Help] [answered]
Hello

I'm using Repeater, and I noticed that any disabled field would not be added to the json output, is there any way to make it save the disabled field? (it will even save values that was in the initial array and not presented in a field)

The reason I'm disabling some fields is that I don't want the user to change them (read-only), and I'm using fields to make it easy to code creation and editing form (I can use the same code)

```
$requestSchema = [
            [
                "name" => "file",
                "type" => "file",
                "required" => true,
            ]
]

Repeater::make('requestSchema')->schema([
                                    TextInput::make('_mandatory')->hidden()->default(false),
                                    TextInput::make('name')->required()->disabled(function (Get $get) { 
                                        return   
                                            (!is_null($get('_mandatory')) ) ? ($get('_mandatory')) : false; 
                                    }),
                                    TextInput::make('type')->required()->disabled(function (Get $get) { 
                                        return   
                                            (!is_null($get('_mandatory')) ) ? ($get('_mandatory')) : false; 
                                    }),
]);

```

The output will be like

```
[
	{
		"required": true,
		"_mandatory": true,
	}
]
```

**Comment by danharrin** [comment, marked as answer]
`dehydrated()`. But beware that its not truly "disabled", the user can still change it via JavaScript.

**Comment by mrahmadt** [comment]
Thanks @danharrin , working fine

I think it should return the values by default, I don't see why I have to dehydrat

**Reply by danharrin** [reply]
1) In v2 it was the other way round and people complained
2) As I said to you, its a security issue, you can still change the field value from JavaScript, that isnt good default behaviour IMO
